src/main/java/org/jenkinsci/plugins/gitlab/GitlabMergeRequestWrapper.java

```diff
@@ -349,6 +350,22 @@ public GitlabNote createNote(String message, boolean shouldClose, boolean should
 
     }
 
+    public GitlabCommitStatus createCommitStatus(String status, String targetUrl) {
+        try {
+            GitlabAPI api = builder.getGitlab().get();
+            GitlabMergeRequest mergeRequest = api.getMergeRequest(project, id);
+            GitlabCommit latestCommit = getLatestCommit(mergeRequest, api);
+
+            if (latestCommit != null) {
+                return api.createCommitStatus(project, latestCommit.getId(), status, mergeRequest.getSourceBranch(), "Jenkins", targetUrl, null);
```

Comment:
Yes, that does make sense. I'll look into getting the commit hash from Jenkins I think as Gitlab does not have that information from their API.

Thanks for the comments!